assets/javascripts/modules/autoCompleteFactory.js

```diff
@@ -2,15 +2,37 @@ require('jquery');
 var autoComplete = require('./autoComplete.js');
 
 var createAutoCompleteCountries = function () {
-  var $chooseCountryAutoCompleteElem = $('.js-choose-country-auto-complete');
-  var $countryCodeInput = $('#countryCode');
-  var suggestionDisplayTemplate = function (title, value) {
-    return title + ' (+' + value + ')';
-  };
+  $('.js-choose-country-auto-complete').each(function(){
+    
+    var $countryCodeInput;
+    if($('#countryCode').length > 0){
+      $countryCodeInput = $('#countryCode');
+    } else {
+      $countryCodeInput = $('[id="' + $(this).attr("data-suggestions") + '"]');    
+    }
+    
+    var suggestionDisplayTemplate = function (title, value) {
+      return title + ' (+' + value + ')';
+    };
+    
+    var suggestions;
+    if($('select[id="' + $(this).attr("data-suggestions") + '"]').length > 0){    
```

Comment:
so you are building suggestions from the target element. This also feels like it shouldn't be in the factory it should be internal in the `autoComplete.js` file. 
So I would make the second argument `suggestions` accept either a string or an object. If a string get that element, if an object do what you currently do. I would also abstract the work of greating suggestion into a high level function for self documentation `createSuggestionsFromSelect` for instance